pints/_mcmc/_population.py

```diff
@@ -275,16 +272,17 @@ def tell(self, fx):
             # Return first point for chain 0
             sample = np.array(self._current[0], copy=False)
             sample.setflags(write=False)
-            return sample
+            return sample, self._current_log_pdfs[0], True
 
         # Perform mutation step (update one chain)
 
         # Update chain, get new sample
-        sample = self._chains[self._i].tell(fx * (1 - self._schedule[self._i]))
+        sample, sample_log_pdf, accepted = self._chains[self._i].tell(
+            fx * (1 - self._schedule[self._i]))
 
-        # Check if accepted
-        if np.any(sample != self._current[self._i]):
-            # Update current sample and untempered log pdf
+        # Update current sample and untempered log pdf
+        if accepted:
+            # Don't use sample_log_pdf here, which is tempered
```

Comment:
Was this the bug?

Reply:
Or it was https://github.com/pints-team/pints/pull/1250/files#diff-dd2521d7bad8361241509fcc5108df3add9b81e36c9d78cb281894a2f4898d7fR329-R330?

Reply:
There were two bugs in pop mcmc itself, both fixed here: https://github.com/pints-team/pints/pull/1246/files

This ticket solves a separate but related bug: the controller couldn't store the accepted sample's log likelihood without these changes to tell